### PR TITLE
Added the ability to set a password for the AP.

### PR DESCRIPTION
**Pull request by ow-breaker**
This PR is also dependent on #11, if that gets rejected I'll have to remove some code from this PR.

### CODE REVIEW

File: src/network.cpp [hunk 1]
```diff
@@ -92,7 +96,11 @@ void setupWifi() {
   WiFi.mode(stationMode ? WIFI_AP_STA : WIFI_AP);
   char apName[64];
   sprintf(apName, "Owie-%04X", ESP.getChipId() & 0xFFFF);
-  WiFi.softAP(apName);
+  if (strcmp(Settings->ap_self_password, "") != 0) {
```
Comment (lolwheel) [inline src/network.cpp:99]: No need for this check, an empty password passed into this function keeps the WiFi open

**Review by lolwheel — COMMENTED** [review]
This leaves a possibility of locking out user out of Owie when they forget/mistype the password.

You'll have to add a codepath to reset all settings in the recovery mode to circumvent that.

File: src/network.cpp (hunk 1 above)
Comment (ow-breaker) [reply src/network.cpp:99]: So it seems like when this check isn't there, the AP name gets all bungled.
Not quite sure why, but my network should be named Owie-C48A, not ESP-DAC48A.
image

Comment (ow-breaker) [reply src/network.cpp:99]: > So it seems like when this check isn't there, the AP name gets all bungled.

Actually, I take this back. The AP name getting bungled and all the following issues seem to be occuring semi-randomly. Investigating now.

File: src/recovery.cpp [hunk 2]
```diff
@@ -29,6 +29,15 @@ void recovery_setup() {
   });
   dnsServer.start(53, "*", WiFi.softAPIP());  // DNS spoofing.
   dnsServer.setErrorReplyCode(DNSReplyCode::NoError);
+  webServer.on("/reset", HTTP_ANY, [](AsyncWebServerRequest *request) {
```
Comment (lolwheel) [inline src/recovery.cpp:32]: This is janky. The OTA library I'm using doesn't allow customization whatsoever so let's just nuke *all* settings whenever we log into recovery mode.

Comment (ow-breaker) [reply src/recovery.cpp]: Lmao, appreciate the compliment. Made a new function in recovery called `nukeSettings()`.

**Pull request merged**